vmdb/app/controllers/api_controller/action.rb

```diff
@@ -0,0 +1,33 @@
+class ApiController
+  module Action
+    private
+
+    def api_action(type, id, options = {})
+      cspec = collection_config[type]
+      klass = cspec[:klass].constantize
+
+      result = yield(klass) if block_given?
+
+      add_href_to_result(result, type, id) if options[:add_href]
```

Comment:
Since the caller is directing this with an option, and it happens immediately after the yield, should this even be in the boilerplate, or should it just be in the caller code.

You have to take the 80/20 rule here.  If 80% of the callers add the href, then you should have an option that ignores the href, and the default would be to add it.  If 20% of the callers don't need the href, then how you have it now is fine.  If it's like 50/50, then the caller should just do it explicitly and don't have the option here at all.